datafusion/physical-plan/src/aggregates/mod.rs

```diff
@@ -1231,6 +1233,13 @@ fn evaluate(
     expr: &[Arc<dyn PhysicalExpr>],
     batch: &RecordBatch,
 ) -> Result<Vec<ArrayRef>> {
+    // handle count() case
+    if expr.is_empty() {
+        return Ok(vec![
+            Arc::new(Int64Array::from(vec![1; batch.num_rows()])) as ArrayRef
```

Comment:
This is equivalent to `count(1)` case

Reply:
It seems that this function is not only used by `count`. I'm not quite sure about the impact of this change.
Ideally, this function should not involve the logic of any specific aggregation function.